app/controllers/ileQuery/IleQueryController.scala

```diff
@@ -0,0 +1,81 @@
+package controllers.ileQuery
+
+import connectors.CustomsDeclareExportsMovementsConnector
+import connectors.exchanges.IleQueryExchange
+import controllers.actions.AuthenticatedAction
+import controllers.exchanges.AuthenticatedRequest
+import forms.IleQuery
+import javax.inject.{Inject, Singleton}
+import models.UcrBlock
+import models.cache.Answers
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, Cookie, DiscardingCookie, MessagesControllerComponents, Result}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.FieldValidator.validDucr
+import views.html.ile_query
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class IleQueryController @Inject()(
+  authenticate: AuthenticatedAction,
+  mcc: MessagesControllerComponents,
+  connector: CustomsDeclareExportsMovementsConnector,
+  ileQueryPage: ile_query
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] = authenticate { implicit request =>
+    Ok(ileQueryPage(IleQuery.form))
+  }
+
+  def submit(): Action[AnyContent] = authenticate { implicit request =>
+    IleQuery.form.fold(
+      formWithErrors => BadRequest(ileQueryPage(formWithErrors)),
+      validUcr => Redirect(controllers.ileQuery.routes.IleQueryController.submitQuery(validUcr))
+    )
+  }
+
+  def submitQuery(ucr: String): Action[AnyContent] = authenticate.async { implicit request =>
```

Comment:
I am still thinking that this method should be either in a separate controller, or to rename `submit` or `submitQuery` to clearly show what they are doing and which part of the journey they conver.
Currently their names could be swapped with no implications.

Reply:
The first two methods are really small, I don't see value to extracting this to the new controller but I can do it.

I'll change the names to be more describing.

Reply:
I'm still confused by the method name.  Only when I got to the routes file did I realised that this is a "GET" endpoint.

Maybe this should be `displayQueryResponse' ?   (I realise that the outcome of calling this end-point is not always the query response - but that is the main intention.